src/util/simplify_expr.cpp

```diff
@@ -2023,11 +2032,11 @@ simplify_exprt::simplify_byte_update(const byte_update_exprt &expr)
           bytes_req = (*val_size) / 8 - val_offset;
 
         byte_extract_exprt new_val(
-          byte_extract_id(),
+          ID_byte_extract_little_endian,
           value,
           from_integer(val_offset, offset.type()),
-          array_typet(unsignedbv_typet(8),
-                      from_integer(bytes_req, offset.type())));
+          array_typet(
+            unsignedbv_typet(8), from_integer(bytes_req, offset.type())));
```

Comment:
I know this a refactor, but can we replace the 8 with something less magic?

Reply:
#917, which this PR is supposed to make easier to merge :-)